clang/lib/Parse/Parser.cpp

```diff
@@ -1224,6 +1225,9 @@ Parser::DeclGroupPtrTy Parser::ParseDeclOrFunctionDefInternal(
 Parser::DeclGroupPtrTy Parser::ParseDeclarationOrFunctionDefinition(
     ParsedAttributes &Attrs, ParsedAttributes &DeclSpecAttrs,
     ParsingDeclSpec *DS, AccessSpecifier AS) {
+  // Add an enclosing time trace scope for a bunch of small scopes with
+  // "EvaluateAsConstExpr".
+  llvm::TimeTraceScope TimeScope("ParseDeclarationOrFunctionDefinition");
```

Comment:
Should we supply a second argument to the ctor here so that we can include the function name as part of the time trace output?

Reply:
Also, I wonder if we want more or less granularity here. This will fire for function declarations and function definitions, so we could add it to `Parser::ParseFunctionDefinition()` instead to only time function definition parsing, or we could split it up so that we time parsing function signatures separately from parsing the function body so we can more easily see how much time is spent in which activity.

Reply:
Good point, thanks. It is added in the commit https://github.com/llvm/llvm-project/pull/65268/commits/fa0503d3b7.